lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -691,98 +725,220 @@ PXR_NS::SdfValueTypeName ufeTypeToUsd(const std::string& ufeType)
     }
 }
 
-PXR_NS::VtValue vtValueFromString(const std::string& typeName, const std::string& strValue)
+VtValue vtValueFromString(const SdfValueTypeName& typeName, const std::string& strValue)
 {
-    PXR_NS::VtValue result;
-    if (typeName == Ufe::Attribute::kBool) {
-        result = "true" == strValue ? true : false;
-    } else if (typeName == Ufe::Attribute::kInt) {
-        result = std::stoi(strValue.c_str());
-    } else if (typeName == Ufe::Attribute::kFloat) {
-        result = std::stof(strValue.c_str());
-    } else if (typeName == Ufe::Attribute::kDouble) {
-        result = std::stod(strValue.c_str());
-    } else if (typeName == Ufe::Attribute::kString) {
-        result = strValue;
-    } else if (typeName == Ufe::Attribute::kEnumString) {
-        result = PXR_NS::TfToken(strValue.c_str());
-    } else if (typeName == Ufe::Attribute::kInt3) {
-        std::vector<std::string> tokens = splitString(strValue, "()[], ");
-        if (TF_VERIFY(tokens.size() == 3, kInvalidValue, strValue.c_str(), typeName.c_str())) {
-            result = GfVec3i(
+    const auto _int3Converter = [](const std::string& s) {
```

Comment:
Unused lambdas. Will remove. They were inlined below since we never repeat them.